File: programming files/datetime_convert.cc
#include <iostream>
#include <iomanip>
#include <sstream>
#include <chrono>

int main()
{
    // dummy date
    std::string date_str = "2022-03-17 10:45:30";
    // create an object of std::tm
    std::tm date_obj = {};
    std::istringstream ss(date_str);
    // parse the date string
    ss >> std::get_time(&date_obj, "%Y-%m-%d %H:%M:%S");
    std::stringstream formatted_date_ss;
    // format the date
    formatted_date_ss << std::put_time(&date_obj, "%m/%d/%Y %H:%M:%S");

    std::string formatted_date = formatted_date_ss.str();

    // print it out
    std::cout << formatted_date << std::endl;

    return 0;
}

File: programming files/datetime_convert.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class datetime_convert {
    public static void main(String[] args) {
        //the date and time as a string
        String dateStr = "2022-03-17 10:45:30";
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");//the format that the date time should take
        LocalDateTime dateObj = LocalDateTime.parse(dateStr, formatter); //parse the format
        String formattedDate = dateObj.format(DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm:ss"));//format the date
        
        System.out.println(formattedDate);//print the formatted date to console
    }
}
